### ISSUE TITLE
auto adjust hp/mana/cap per level like CIP

### ISSUE THREAD

**Issue by jerryb1988** [otland/forgottenserver#2194]
I'm just submitting this as a feature/improvement. I modded login.lua and added the following code before registering the events.

```
	-- Fix HP/Mana/Cap to match CIP, assume player left rook at level 8
	if player:getVocation():getId() == 0 then
		
		calchp = 5 * (player:getLevel() + 29)
		calcmana = 5 * (player:getLevel() + 10)
		calccap = 10 * (player:getLevel() + 39)
		
	elseif player:getVocation():getId() == 1 or player:getVocation():getId() == 5 then
		
		calchp = 5 * (player:getLevel() + 29)
		calcmana = 5 * ((6 * player:getLevel()) - (5 * 8) + 10)
		calccap = 10 * (player:getLevel() + 39)
		
	elseif player:getVocation():getId() == 2 or player:getVocation():getId() == 6 then
		
		calchp = 5 * (player:getLevel() + 29)
		calcmana = 5 * ((6 * player:getLevel()) - (5 * 8) + 10)
		calccap = 10 * (player:getLevel() + 39)
		
	elseif player:getVocation():getId() == 3 or player:getVocation():getId() == 7 then
		
		calchp = 5 * ((2 * player:getLevel()) - 8 + 29)
		calcmana = 5 * ((3 * player:getLevel())- (2 * 8) + 10)
		calccap = 10 * ((2 * player:getLevel()) - 8 + 39)
		
	elseif player:getVocation():getId() == 4 or player:getVocation():getId() == 8 then
		
		calchp = 5 *((3 * player:getLevel()) - (2 * 8) + 29)
		calcmana = 5 *(player:getLevel() + 10)
		calccap = 5 *((5 * player:getLevel()) - (5 * 8) + 94)
		
	end	

```

These formulas were taken from the tibia wiki. They assume that for any character with a vocation, that they left rookgaard at level 8 (so there will be no handicap). Their may be a better way to do this via hardcoding, but this was a quick and easy solution to characters having incorrect values when being bumped levels from the start. I realize this will trigger unnecessarily for every login and could potentially create a performance hit on a large server, so this could also be something that could become a talk action like !fixme or something of the sorts. Feel free to incorporate however you deem it fits.

**Comment by DSpeichert** [comment]
This is a handy script. Some people would love to have it while others may intentionally have different values they don't want overwritten.

I think this could be added with a comment in the datapack that it's safe to delete/disable. The reason why there is no point in disabling it by default is that nobody with an existing datapack would add it anyway.

I don't think there's significant performance problem here, it fires per login and operates on already preloaded values of a single player.

**Comment by jerryb1988** [comment]
@DSpeichert Forgive me as I am new to GitHub. I'm not sure how to properly submit a pull request.

**Comment by EPuncker** [comment]
hey @DSpeichert I posted the script on otland [here](https://otland.net/threads/tfs-1-x-auto-adjust-hp-mana-cap-per-level-like-cip.268336/) with the due credit to @jerryb1988 , you may want to close this issue maybe

**Comment by Znote** [comment]
I agree, while cool, this probably shouldn't be added as a PR to the repo. Some servers may intetionally want to differentiate this, and may use other default vocation gain values.

**Comment by DSpeichert** [comment]
IMHO it SHOULD be added but as another login script that maybe is disabled. Can we disable scripts in creaturescripts? Or just keep it commented out. There's value in shipping that code.

**Comment by WibbenZ** [comment]
We can't disable it, we can only disable events?

But might be a good thing to add IMO, maybe to all classes?

Haven't tested it but I agree that it should be merged with it comment out in XML or adding enabled and setting it to false / 0

**Comment by Znote** [comment]
Ohh, we can add it, but disable it as a RevScriptSys creaturescript by adding a # symbol in front of the filename.

**Comment by EvilHero90** [comment]
I don't really like this idea tbh, even if it is disabled by default.
Imagine having items which give max HP/Mana through a lua script, this script would heavily cause problems then and unless you relog and re equip items etc. it'll keep messing with it.
This is not really something we should have for a vanilla server.

**Comment by Zbizu** [comment]
The idea is good but it should get values from Vocations class instead of being hardcoded into the script.
